include/libu64/gfxprint.h

```diff
@@ -3,33 +3,105 @@
 
 #include "ultra64.h"
 #include "color.h"
+#include "libc64/aprintf.h"
 
-typedef struct gfxprint {
-    /* 0x00 */ char unk00[0x30];
-} gfxprint; // size = 0x30
-
-// void gfxprint_setup();
-void gfxprint_color(gfxprint* this, u8 r, u8 g, u8 b, u8 a);
-// void gfxprint_locate();
-void gfxprint_locate8x8(gfxprint* this, s32 arg1, s32 arg2);
-// void gfxprint_setoffset();
-// void gfxprint_putc1();
-// void gfxprint_putc();
-// void gfxprint_write();
-// void gfxprint_puts();
-// void gfxprint_prout();
+typedef struct gfxprint  {
+    /* 0x00 */ PrintCallback proutFunc; /* Current print out func */
+    /* 0x04 */ Gfx* gListp; /* Current display list to write text to */
+    /* 0x08 */ u16 positionX; /* Display coords (sub-pixel units) */
+    /* 0x0A */ u16 positionY; /* Display coords (sub-pixel units) */
+    /* 0x0C */ u16 offsetX; /* Display offset */
+    /* 0x0E */ u8 offsetY; /* Display offset */
+    /* 0x0F */ u8 flags; /* 1: Kana mode -> 0: Katakana/1: Hiragana
+                 2: Gradient displayed
+                 4: Shadow displayed
+                 8: Attribute changed
+                 64: Low-res -> high-res conversion
+                 128: Opened state */
```

Comment:
The flags are all pretty well documented below. Not sure all the flag comments are needed here.